packages/lib/passport-interface/src/SubscriptionManager.ts

```diff
@@ -61,15 +61,102 @@ export class FeedSubscriptionManager {
   public constructor(
     api: IFeedApi,
     providers?: SubscriptionProvider[],
-    activeSubscriptions?: Subscription[]
+    activeSubscriptions?: Subscription[],
+    errors?: Map<string, SubscriptionError>
   ) {
     this.updatedEmitter = new Emitter();
     this.api = api;
     this.providers = providers ?? [];
     this.activeSubscriptions = (activeSubscriptions ?? []).map(ensureHasId);
-    this.errors = new Map();
+    this.errors = errors !== undefined ? new Map(errors) : new Map();
   }
 
+  /**
+   * Creates a new FeedSubscriptionManager with the data from this one.
+   *
+   * This includes all data (subscriptions, providers, and errors, but doesn't
+   * include dynamic state (listeners on the emitter).
+   *
+   * This is a shallow clone.  The resulting object has new containers but
+   * the same underlying objects, which are expected to be immutable.  If you
+   * need a deep clone, use serialization.
+   */
+  public clone(): FeedSubscriptionManager {
```

Comment:
fwiw generally i prefer to implement `clone` via a `serialize`/`deserialize` - saves a few implementation headaches, eg. having to test the new clone functionality separately from ser/de. generally i am also fine paying the perf cost of that. don't mind that you've done it this way here.

Reply:
This is a shallow clone for performance reasons.  The docs above say how to get a deep clone.
Mind you, this didn't end up being used in merge, outside of tests, so it could just get deleted.
I'm going to keep it for now in case of future tweaks to the merge or mutation algorithms.
I'd like to move toward less "modify in place" and more "copy on write", though for the merge I ended up with, that was done implicitly by the fact that the remote collections are always freshly deserialized